api/controllers/UserController.js

```diff
@@ -2,41 +2,61 @@ import jwt from 'jsonwebtoken'
 import { getPublicKeyFromPem } from '../../lib/util'
 
 module.exports = {
-  create: function (req, res) {
-    const { name, email, email_validated, password } = req.allParams()
 
-    return User.create({name, email: email ? email.toLowerCase() : null, email_validated, account: {type: 'password', password}})
-    .then(async (user) => {
-      await Analytics.trackSignup(user.id, req)
-      if (req.param('login')) {
-        await UserSession.login(req, user, 'password')
+  create: async function (req, res) {
+    const { name, email, groupId } = req.allParams()
+    const group = groupId && await Group.find(groupId)
+
+    let user = await User.find(email, {}, false)
+    if (user) {
+      // User already exists so just invite them to the new group (if there is one)
+      if (group) {
+        // TODO: create Invitation object and send, but from what user? or do we add them directly to the group? but then they start receiving emails without consenting...
```

Comment:
How do we know again that this user has permissions to be part of the group at `groupId`?  Could just any `groupId` be sent along with a `UserController#create` call and creating an invitation?

Reply:
yeah... this is why this is a super hacky MVP of the API, and why i created "super" roles for these API clients, they can add users to any group right now